apps/mitt-konto/src/MittKonto/Main.purs

```diff
@@ -143,347 +100,24 @@ render self@{ state, setState } logger =
           , render: const mittKontoView
           }
 
-loadingIndicator :: Loading -> JSX
-loadingIndicator Loading =
-  DOM.div
-      { className: "mitt-konto--loading flex items-center"
-      , children:
-          [ DOM.div
-              { className: "clearfix mx-auto"
-              , children: [ DOM.div { className: "mitt-konto--loading-image", children: [] } ]
-              }
-          ]
-      }
-
--- | Allows to run the asynchronous action while showing the loading indicator
---   and handling the result.
-withSpinner :: forall a. (Maybe Loading -> Effect Unit) -> Aff a -> Aff a
-withSpinner setLoadingState action = do
-   let timeoutDelay = Aff.Milliseconds $ 30.0 * 1000.0
-       flickerDelay = Aff.Milliseconds $ 200.0
-   -- The "loading" thread turns the spinner on (when started) and off (when killed).
-   -- Prevent the spinner from flickering.
-   loadingFiber <-
-     Aff.forkAff $ (do
-       -- delay turning on the spinner, in case if the action is "instantanious"
-       Aff.delay flickerDelay
-       -- invincibly sleep for a bit more (would still wait if killed here)
-       Aff.invincible $ do
-         -- turn the spinner on
-         liftEffect $ setLoadingState $ Just Loading
-         Aff.delay flickerDelay
-       -- in the end we sleep indefinetely. When killed, remove the spinner
-       Aff.never) `Aff.cancelWith` Aff.effectCanceler (setLoadingState Nothing)
-   -- The "action" thread runs the asynchronous task
-   actionFiber <- Aff.forkAff action
-   -- The "timeout" thread would kill the "action" thread after the delay.
-   timeoutFiber <- Aff.forkAff do
-     Aff.delay timeoutDelay
-     Aff.killFiber (error "Timeout reached") actionFiber
-     Aff.killFiber (error "Timeout reached") loadingFiber
-   Aff.joinFiber actionFiber # Aff.finally do
-     -- finally in the end, when the action has been completed
-     -- we kill all other threads and switch the loading off
-     Aff.killFiber (error "Action is done") timeoutFiber
-     Aff.killFiber (error "Action is done") loadingFiber
-
 -- | Navbar with logo, contact info, logout button, language switch, etc.
-navbarView :: Self -> Sentry.Logger -> JSX
+navbarView :: Types.Self -> Sentry.Logger -> JSX
```

Comment:
How about moving the rest to the Views directory as well? Clump the minor ones into a Misc.purs or something.

Reply:
How about instead of Misc.purs, move them to the main Views.purs directly?

Reply:
I think I prefer Valtteri's suggestion! But In any case I was bothered too by having these remaining views in `Main.purs`.

> Modules named like Misc or Helpers or something gives me the creeps for some reason 🙀 They are usually some sort of dumpsters

Yes I agree in general, but I have to say that in this case I couldn't find a better alternative. I don't know sometimes I think that how **React** is made makes it difficult to modularize code 🤔

Reply:
> Yes I agree in general, but I have to say that in this case I couldn't find a better alternative.

I was talking about the `Helpers` module there

Reply:
Yeah, it's fine 🙂